packages/ai/src/models/chat.ts

```diff
@@ -2,16 +2,16 @@ import { Function } from '../function';
 import { Memory } from '../memory';
 import { Message, ModelMessage, SystemMessage, UserMessage } from '../message';
 
-export interface ChatParams {
+export type ChatParams<TExtraParams extends {}> = {
```

Comment:
I like the direction, but since we are still in alpha I suggest going with the breaking change and making this typing a bit cleaner. I think we should sit down and figure out the ideal interface for this since once it goes live breaking changes will be more difficult.